src/web/teap/teap/nextcloud/api.py

```diff
@@ -0,0 +1,52 @@
+from flask import Blueprint, jsonify, request, abort
+from flask.views import MethodView
+
+from nextcloud import NextCloud
+
+blueprint = Blueprint('nextcloud_api', __name__, url_prefix='/api')
+
+
+class NextCloudMixin:
+
+    @property
+    def nextcloud(self):
+        """ Get nextcloud instance """
+        # tmp mock nextcloud credentials
+        url = 'http://localhost:8080'
```

Comment:
Obviously, these values are candidates to make configurable e.g. by using env variables.

Reply:
I think it will be attached to user somehow. As I understand platform can be used for different nextcloud instances, or it will be deployed for the only one?

Reply:
Great question! The current use case is one instance. Imagine an enterprisey setup featuring e.g. Nextcloud, Rocket.Chat, Gitlab, etc. - you will have one instance for each (at least at the beginning), but at the same time there will be those multiple services that have to be managed.
The idea is that the service will have admin access, as only admins will be allowed to get inside.

Reply:
Thanks for clarification! I moved values to config. 
We will probably need to attach username, password for every user separately in the future, but for now left it in the config as well.

Off topic: previously I thought that it should be some kind of service where anybody can register, add their own instances of Nextcloud, Gitlab, whatever, and manage them. But it probably looks more organic as a part of enterprise build, not as SaaS.